### PR TITLE
Migrate Relationship Type to Search kit 

### PR DESCRIPTION
**Pull request by Damilare1**
Overview
----------------------------------------
Add relationship type files, update paths, and add form templates for Relationship Types to civicrm_admin_ui ang directory.

Before
----------------------------------------
Relationship Type
Screenshot_2022-10-10_09-40-31

After
----------------------------------------
Relationship Type
Screenshot_2022-10-10_09-41-23_new

Technical Details
----------------------------------------
Created the new path in CRM/Core/xml/Menu/Admin.xml in order to prevent the actions from redirecting to the same base page.

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins add to whitelist

File: CRM/Contact/DAO/RelationshipType.php [hunk 1]
```diff
@@ -30,6 +30,18 @@ class CRM_Contact_DAO_RelationshipType extends CRM_Core_DAO {
...
+  /**
+   * Paths for accessing this entity in the UI.
+   *
+   * @var string[]
+   */
+  protected static $_paths = [
+    'add' => 'civicrm/admin/reltype/edit?action=add&reset=1',
+    'view' => 'civicrm/admin/reltype/edit?action=view&id=[id]&reset=1',
+    'update' => 'civicrm/admin/reltype/edit?action=update&id=[id]&reset=1',
+    'delete' => 'civicrm/admin/reltype/edit?action=delete&id=[id]',
```
Comment (colemanw) [inline CRM/Contact/DAO/RelationshipType.php:42]: We should add `&reset=1` to the end of the delete path.

File: ext/civicrm_admin_ui/ang/afsearchRelationshipTypes.aff.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="af-container">
+  <div class="af-markup">
+    <div class="help">
+    <p>{{:: ts(&#39;Relationship types describe relationships between people, households and organizations. Relationship types labels describe the relationship from the perspective of each of the two entities (e.g. Parent &gt;-&lt; Child, Employer &gt;-&lt; Employee). For some types of relationships, the labels may be the same in both directions (e.g. Spouse &gt;-&lt; Spouse).&#39;) }} <a class="crm-doc-link no-popup" href="https://docs.civicrm.org/user/en/latest/organising-your-data/relationships" target="_blank">{{:: ts(&#39;(Learn more...)&#39;) }}</a></p>
+
+    <p>{{:: ts(&#39;You can define as many additional relationships types as needed to cover the types of relationships you want to track. Once a relationship type is created, you may also define custom fields to extend relationship information for that type from &#39;) }} <a href="civicrm/admin/custom/group?reset=1">{{:: ts(&#39;Administer CiviCRM &raquo; Custom Data&#39;) }}</a></p>
```
Comment (colemanw) [inline ext/civicrm_admin_ui/ang/afsearchRelationshipTypes.aff.html:6]: I don't think escaping straight-quotes with `&#39;` is necessary, and it may prevent i18n string parsing

File: ext/civicrm_admin_ui/ang/afsearchRelationshipTypes.aff.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "is_public": false,
+    "is_token": false,
+    "contact_summary": null,
+    "icon": "fa-list-alt",
+    "server_route": "civicrm/admin/reltype",
+    "permission": "access CiviCRM",
+    "redirect": null,
+    "create_submission": false,
+    "navigation": null
+}
```
Comment (colemanw) [inline ext/civicrm_admin_ui/ang/afsearchRelationshipTypes.aff.json:18]: Needs newline at eof.

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Relationship_Types.mgd.php [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+              'path' => 'civicrm/admin/reltype/edit?action=add&reset=1', 
+              'text' => 'Add Relationship Type', 
+              'icon' => 'fa-plus',
+            ],
+          ], 
+          'acl_bypass' => FALSE,
+        ],
+      ],
+    ],
+  ];
```
Comment (colemanw) [inline ext/civicrm_admin_ui/managed/SavedSearch_Administer_Relationship_Types.mgd.php:154]: Also missing eof newline

**Review by colemanw — CHANGES_REQUESTED** [review]
@Damilare1 overall this looks like a great 1st PR. I left a couple code comments. After you tweak the path in the xml you'll need to re-run `setup.sh -g` and also please remove changes to `package.lock` from this PR they probably snuck in by accident when you ran `setup`.

Also be sure to re-run `civilint` before committing - the style checker flagged a bunch of extra eol whitespace.

File: ext/civicrm_admin_ui/ang/afsearchRelationshipTypes.aff.html (hunk 2 above)
Comment (Damilare1) [reply ext/civicrm_admin_ui/ang/afsearchRelationshipTypes.aff.html]: True. This was auto-generated during creation of the form, I've replaced it in the new commit.

**Review by colemanw — APPROVED** [review]

File: CRM/Admin/Form/Options.php [hunk 5]
```diff
@@ -50,6 +50,9 @@ class CRM_Admin_Form_Options extends CRM_Admin_Form {
    */
   public function preProcess() {
     parent::preProcess();
+    CRM_Utils_Request::retrieve('action', 'String', $this);
+    CRM_Utils_Request::retrieve('id', 'Positive', $this, FALSE, 0);
+    $this->set('BAOName', 'CRM_Core_BAO_OptionValue');
```
Comment (colemanw) [inline CRM/Admin/Form/Options.php:55]: I don't think this change belongs in this PR.

File: CRM/Admin/Page/Options.php [hunk 6]
```diff
@@ -152,6 +152,12 @@ public function getBAOName() {
   public function &links() {
     if (!self::$_links) {
       self::$_links = [
+        CRM_Core_Action::VIEW => [
+          'name' => ts('View'),
+          'url' => 'civicrm/admin/options/' . self::$_gName,
+          'qs' => 'action=view&id=%%id%%&reset=1',
+          'title' => ts('View Relationship Type'),
+        ],
```
Comment (colemanw) [inline CRM/Admin/Page/Options.php:160]: @Damilare1 this file looks unrelated.

File: CRM/Core/xml/Menu/Admin.xml [hunk 7]
```diff
@@ -580,7 +585,7 @@
   <item>
      <path>civicrm/admin/options/grant_type</path>
      <title>Grant Types</title>
-     <desc>List of types which can be assigned to Grants. (Enable CiviGrant from Administer > Systme Settings > Enable Components if you want to track grants.)</desc>
+     <desc>List of types which can be assigned to Grants. (Enable CiviGrant from Administer > System Settings > Enable Components if you want to track grants.)</desc>
```
Comment (colemanw) [inline CRM/Core/xml/Menu/Admin.xml:588]: @Damilare1 this change belongs in the other PR

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Relationship_Types.mgd.php [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+                  'condition' => [],
+                ],
+              ],
+              'type' => 'buttons',
+              'alignment' => 'text-right',
+            ],
+          ],
+          'addButton' => [
+            'path' => 'civicrm/admin/reltype/edit?action=add&reset=1',
+            'text' => 'Add Relationship Type',
```
Comment (colemanw) [inline ext/civicrm_admin_ui/managed/SavedSearch_Administer_Relationship_Types.mgd.php:143]: Needs `E::ts()`

**Comment by Damilare1** [comment]
@colemanw do you have any other comments on this?

**Comment by colemanw** [comment]
I checked this again. It looks perfect. Great work @Damilare1
